yarn-project/noir-libs/noir-aztec/src/abi.nr

```diff
@@ -132,27 +132,27 @@ struct HistoricBlockData {
     contract_tree_root : Field,
     l1_to_l2_messages_tree_root : Field,
     blocks_tree_root: Field,
-    prev_global_variables_hash: Field,
     public_data_tree_root: Field,
+    global_variables_hash: Field,
 }
 
 impl HistoricBlockData {
     // NOTE: this order must match the order in `private_circuit_public_inputs.hpp`
-    fn serialize(self) -> [Field; CONSTANT_HISTORIC_BLOCK_DATA_LENGTH] {
+    fn serialize(self) -> [Field; HISTORIC_BLOCK_DATA_LENGTH] {
         [
             self.private_data_tree_root,
             self.nullifier_tree_root,
             self.contract_tree_root,
             self.l1_to_l2_messages_tree_root,
             self.blocks_tree_root,
-            self.prev_global_variables_hash,
-            self.public_data_tree_root
+            self.public_data_tree_root,
+            self.global_variables_hash,
         ]
     }
 }
 
 fn empty_block_data() -> HistoricBlockData {
-    HistoricBlockData{ private_data_tree_root: 0, nullifier_tree_root: 0, contract_tree_root: 0, l1_to_l2_messages_tree_root: 0, blocks_tree_root: 0, prev_global_variables_hash: 0, public_data_tree_root: 0 }
+    HistoricBlockData{ private_data_tree_root: 0, nullifier_tree_root: 0, contract_tree_root: 0, l1_to_l2_messages_tree_root: 0, blocks_tree_root: 0, public_data_tree_root: 0,  global_variables_hash: 0 }
```

Comment:
I think we should replace this with an `HistoricBlockData::empty()` instead? Seems better for consistency. Only used in `context.call_public_function_with_packed_args`.